app/com/arpnetworking/metrics/portal/reports/impl/chrome/DefaultDevToolsFactory.java

```diff
@@ -48,13 +54,35 @@ public DevToolsService create(final boolean ignoreCertificateErrors, final Map<S
     /**
      * Public constructor.
      *
-     * @param chromePath is the path to the Chrome binary to use.
+     * @param config is a config that has the following fields: <ul>
+     *   <li>{@code path} points to the Chrome executable</li>
+     *   <li>
+     *     {@code args} is a map of command-line flags passed to Chrome.
+     *     List of valid args: https://peter.sh/experiments/chromium-command-line-switches/
+     *     Keys are flags without the leading {@code --};
+     *     values are strings (for flags that take args) or {@code true} (for flags that don't).
+     *     For example, {@code foo=true, bar="baz"} would result in the Chrome invocation {@code chromium --foo --bar=baz}.
+     *   </li>
+     *   <li>{@code executor} is a sub-object with the fields: <ul>
+     *     <li>{@code corePoolSize} and {@code maximumPoolSize}, which map straightforwardly to
+     *       {@link ThreadPoolExecutor#ThreadPoolExecutor} arguments;</li>
+     *     <li>{@code keepAlive} is an ISO-8601 duration that maps straightforwardly to the same constructor;</li>
+     *     <li>{@code queueSize} is how large the executor's queue should be before task-submissions start blocking.</li>
+     *     </ul>
+     *   </ul>
+     *
+     * TODO(spencerpearson): I don't like exposing the threadpool implementation details, but I can very easily imagine the user
+     *   wanting to configure them.
      */
-    public DefaultDevToolsFactory(final String chromePath) {
-        _chromePath = chromePath;
+    @Inject
+    public DefaultDevToolsFactory(final Config config) {
+        _chromePath = config.getString("path");
+        _chromeArgs = ImmutableMap.copyOf(config.getObject("args").unwrapped());
+        _executor = createExecutorService(config.getConfig("executor"));
```

Comment:
If no `executor` block exists, this will return `null` which even with defaults below I believe would cause an NPE.